src/dev-utils/deltareplayer.js

```diff
@@ -99,24 +104,28 @@ export default class DeltaReplayer {
 	 */
 	applyAllDeltas() {
 		return this.applyNextDelta()
-			.then( () => this.applyAllDeltas() )
-			.catch( () => {} );
+			.then( ( isFinished ) => {
+				if ( !isFinished ) {
+					return this.applyAllDeltas();
+				}
+			} );
 	}
 
 	/**
 	 * Applies the next delta to replay.
+	 * Returns promise with a `isFinished` parameter.
 	 *
-	 * @returns {Promise}
+	 * @returns {Promise.<Boolean>}
```

Comment:
Yes, we do, e.g. here: https://github.com/ckeditor/ckeditor5-image/blob/0e8c6cab3759f5616f6d02a9ffc31aa099491adc/src/imagetextalternative.js#L108. The generic types are documented here: http://usejsdoc.org/tags-type.html, I didn't create the separate doc about them (maybe I should (?)). They're building correctly. And here is a list of currently supported generic types: https://github.com/ckeditor/ckeditor5-dev/blob/0aa523b441a052a441005041af9df4b7bd493c0e/packages/jsdoc-plugins/lib/validator/types.js